src/flows/context/editor.tsx

```diff
@@ -0,0 +1,183 @@
+import React, {
+  FC,
+  createContext,
+  useState,
+  useCallback,
+  useMemo,
+  useContext,
+  useRef,
+  useEffect,
+} from 'react'
+import {EditorType} from 'src/types'
+import {PipeContext} from 'src/flows/context/pipe'
+
+export enum InjectionType {
+  OnOwnLine = 'onOwnLine',
+  SameLine = 'sameLine',
+}
+
+interface InjectionPosition {
+  row: number
+  column: number
+  shouldInsertOnNextLine: boolean
+}
+
+interface InjectionOptions {
+  header?: string | null
+  text: string
+  type: InjectionType
+}
+
+export interface EditorContextType {
+  editor: EditorType | null
+  register: (editor: EditorType) => void
```

Comment:
this function isn't being used?

Reply:
register gets imported and renamed to `setEditorInstance`:
https://github.com/influxdata/ui/blob/a492f43537efa08e248ce662ca0dafd122607b34/src/flows/pipes/RawFluxEditor/view.tsx#L54

which then is the callback handed to the monaco editor:
https://github.com/influxdata/ui/blob/a492f43537efa08e248ce662ca0dafd122607b34/src/flows/pipes/RawFluxEditor/view.tsx#L135

Reply:
can you remove the renaming in line 54 and call the function directly?

Reply:
we also destruct the sidebar `register()` function. So one of the contexts, sidebar or editor, needs the register to be renamed on destructing. Or directly all it.

Reply:
instead, directly passing out `setEditor` prop.